src/test/java/com/epam/jpop/userms/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package com.epam.jpop.userms.service.impl;
+
+import com.epam.jpop.userms.entity.User;
+import com.epam.jpop.userms.exception.UserAlreadyExists;
+import com.epam.jpop.userms.exception.UserNotFound;
+import com.epam.jpop.userms.model.UserDetail;
+import com.epam.jpop.userms.repository.UserRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceImplTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    @Test
+    void getAllUsers() {
+        List<User> users = prepareUserMockData();
```

Comment:
//given
`<code here>`
//when
`<code here>`
//then
`<code here>`

Follow this pattern

Reply:
resolved